cmd/kubeadm/app/cmd/join.go

```diff
@@ -57,12 +58,31 @@ func NewCmdJoin(out io.Writer) *cobra.Command {
 		"(required) Shared secret used to secure bootstrap. Must match the output of 'kubeadm init'",
 	)
 
+	cmd.PersistentFlags().BoolVar(
+		&skipChecks, "skip-checks", false,
+		"skip checks normally run before modifying the system",
+	)
+
 	return cmd
 }
 
 // RunJoin executes worked node provisioning and tries to join an existing cluster.
-func RunJoin(out io.Writer, cmd *cobra.Command, args []string, s *kubeadmapi.NodeConfiguration) error {
+func RunJoin(out io.Writer, cmd *cobra.Command, args []string, s *kubeadmapi.NodeConfiguration, skipChecks bool) error {
 	// TODO(phase1+) this we are missing args from the help text, there should be a way to tell cobra about it
+	if u, ok := preflight.IsRoot(); !ok {
```

Comment:
somewhere we should think about how to minimize duplication between `RunInit` and `RunJoin`

not required to do it now, but anyway